jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/GemUtils.groovy

```diff
@@ -156,6 +156,70 @@ class GemUtils {
         extractGems(project,jRubyClasspath,project.files(gemConfig.files),destDir,action)
     }
 
+    static void setupJars(Configuration config,
+                          File destDir,
+                          GemUtils.OverwriteAction overwrite) {
+        File jarsDir = new File(destDir, 'jars')
+        def artifacts = config.resolvedConfiguration.resolvedArtifacts
+        def fileRenameMap = [:]
+        def coordinates = []
+        def files = []
+        artifacts.each { dep ->
+
+            def group = dep.moduleVersion.id.group
+            def groupAsPath = group.replace('.' as char, File.separatorChar)
+            def version = dep.moduleVersion.id.version
+            // TODO classifier
+            def newFileName = "${groupAsPath}/${dep.name}/${version}/${dep.name}-${version}.${dep.type}"
+
+            if (group != 'rubygems' && dep.type != 'gem' && dep.name != 'jruby-complete') {
+                // TODO classifier and system-scope
+                coordinates << "${group}:${dep.name}:${version}:runtime:"
+            }
+            fileRenameMap[dep.file.name] = newFileName
+            // TODO omit system-scoped files
+            files << dep.file
+        }
+
+        // create Jars.lock file used by jar-dependencies
+        File jarsLock = new File(destDir, 'Jars.lock')
```

Comment:
IMO L185-L200 should be broken out into its own method which would be easier to write unit tests for, e.g.

```groovy
/**
 * @return boolean true if the Jars.lock file has been successfully written
 * @throws DuplicateFileCopyingException if there is an existing Jars.lock file
 */
static boolean writeJarLockFile(List<String> dependencyCoordinates, GemUtils.OverrideAction overwrite)
```


Then the unit tests for the method can simply verify those three behaviors in isolation